libs/header/p2p/session.go

```diff
@@ -139,30 +145,47 @@ func (s *session[H]) doRequest(
 
 	r, size, duration, err := sendMessage(ctx, s.host, stat.peerID, s.protocolID, req)
 	if err != nil {
-		log.Errorw("requesting headers from peer failed", "failed peer", stat.peerID, "err", err)
-		select {
-		case <-s.ctx.Done():
-		case s.reqCh <- req:
-			stat.decreaseScore()
-			log.Debug("retrying the request from different peer")
-		}
-		return
+		// we should not punish peer at this point and should try to parse responses, despite that error was received.
+		log.Debugw("requesting headers from peer failed", "failed peer", stat.peerID, "err", err)
 	}
 
 	h, err := s.processResponse(r)
 	if err != nil {
 		switch err {
```

Comment:
just switch on err here

Reply:
No. It will not work as expected then. In case we are receiving an error, we have to check the error and then return the request to the process loop, to re-request from another peer.

Reply:
can you at least move the `errNoResponses` comment to the proper case statement and fix its name to match the err name?